example/lib/interactive_features_example.dart

```diff
@@ -0,0 +1,92 @@
+  @override
+  final String? subtitle = 'Click to select buildings, long-click to unselect';
+
+  @override
+  State<StatefulWidget> createState() => InteractiveFeaturesState();
+}
+
+class InteractiveFeaturesState extends State<InteractiveFeaturesExample> {
+  InteractiveFeaturesState();
+  MapboxMap? mapboxMap;
+
+  _onMapCreated(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    mapboxMap.style;
+  }
+
+  _onTap(context) async {
+    // Define the geometry to query, in this case the point where the user clicked.
+    var clicked = await mapboxMap?.pixelForCoordinate(context.point);
+    var renderedQueryGeometry =
+        RenderedQueryGeometry.fromScreenCoordinate(clicked!);
+
+    // Define the featureset to query. In this case "buildings", which is defined in the
+    // Standard Experimental style.
+    var featureset =
+        FeaturesetDescriptor(featuresetId: "buildings", importId: "basemap");
+
+    // Query the featureset for the geometry
+    var queriedFeatures = await mapboxMap?.queryRenderedFeaturesForFeatureset(
+        geometry: renderedQueryGeometry, featureset: featureset);
+    var featuresetFeature = queriedFeatures?.first;
+
+    if (featuresetFeature != null) {
+      // Define the state to set for the feature, in this case highlighting
+      // Set that featurestate on that featuresetFeature
+      Map<String, Object?> state = {
+        "highlight": true,
+      };
+      mapboxMap?.setFeatureStateForFeaturesetFeature(featuresetFeature, state);
+    }
+  }
+
+  _onLongTap(context) async {
+    // Define the geometry to query, in this case the point where the user clicked.
+    var clicked = await mapboxMap?.pixelForCoordinate(context.point);
+    var renderedQueryGeometry =
+        RenderedQueryGeometry.fromScreenCoordinate(clicked!);
+
+    // Query the featureset for the geometry
+    var featureset =
+        FeaturesetDescriptor(featuresetId: "buildings", importId: "basemap");
+    var queriedFeatures = await mapboxMap?.queryRenderedFeaturesForFeatureset(
+        geometry: renderedQueryGeometry, featureset: featureset);
+    var featuresetFeatureId = queriedFeatures?.first.id;
+
+    // Remove that feature state
+    if (featuresetFeatureId != null) {
+      mapboxMap?.removeFeatureStateForFeaturesetFeatureDescriptor(
+          featureId: featuresetFeatureId, featureset: featureset);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        body: MapWidget(
+      key: ValueKey("mapWidget"),
+      cameraOptions: CameraOptions(
+          center: Point(coordinates: Position(24.9453, 60.1718)),
+          bearing: 49.92,
+          zoom: 16.35,
+          pitch: 40),
+
+      /// DON'T USE Standard Experimental style in production, it will break over time.
+      /// Currently this feature is in preview.
+      styleUri: MapboxStyles.STANDARD_EXPERIMENTAL,
+      textureView: true,
+      onMapCreated: _onMapCreated,
+      onTapListener: _onTap,
```

Comment:
can I add more than one tap listeners?

Reply:
@pjleonard37 Did you try to implement the `addInteraction` method? I think it makes sense to start with it, instead of QRF. 

With introduction of interactions, the Interaction is the main way to handle taps and we consider the qrf to be the low-level API. Users also will benefit from Interactions more, than from the qrf, because now they have a good cross-platform event handling system.

I think we should start with interactions and promote them to use Interactions in the examples.

Reply:
> @pjleonard37 Did you try to implement the addInteraction method? I think it makes sense to start with it, instead of QRF.

I've experimented a bit locally, but wanted to keep the scope of this PR contained so I started with the low-level APIs. I do agree we should promote Interactions as the main entry point in our examples and docs.

Reply:
Should we create a separate ticket for that, so we don't forget? In the https://mapbox.atlassian.net/browse/MAPSFLT-232 the `addInteraction` was in the scope.

Reply:
Yes -- that ticket should cover `Map.addInteraction`, `TapInteraction`, `LongPressInteraction`, and generated derivative features for Standard Style (like `StandardBuilding`, `StandardPoiLabel`), which I'll create. As for the work itself, as we will not be able to merge this PR into main until a new iOS maps with [this](https://github.com/mapbox/mapbox-maps-ios-internal/pull/2344) lands anyways, it will be in a new PR into this branch.